Intermittent CI failures building distributions

We are having some intermittent CI failures in PRs when building distributions. 

They look like this:

> Run make ci
>   make ci
>   shell: /usr/bin/bash -e {0}
>   env:
>     GOTOOLCHAIN: local
>     DISTRIBUTIONS: otelcol-contrib
> Artifacts to generate: otelcol-contrib,builder,opampsupervisor
> Fetching builder-config.yaml...
> Verifying all manifest.yaml files in 'distributions'...
> Checking distributions/otelcol/manifest.yaml
> Checking distributions/otelcol-k8s/manifest.yaml
> Checking distributions/otelcol-contrib/manifest.yaml
> ./scripts/validate-components.sh: line 73: printf: write error: Broken pipe
> ./scripts/validate-components.sh: line 73: printf: write error: Broken pipe
> ./scripts/validate-components.sh: line 73: printf: write error: Broken pipe
> Checking distributions/otelcol-otlp/manifest.yaml
> Checking distributions/otelcol-ebpf-profiler/manifest.yaml
> 
> The following components MUST be listed in
> https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/cmd/otelcontribcol/builder-config.yaml
> to ensure that they can be built:
> 
> github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sumologicexporter
> github.com/open-telemetry/opentelemetry-collector-contrib/exporter/tencentcloudlogserviceexporter
> github.com/open-telemetry/opentelemetry-collector-contrib/extension/awsproxy
> 
> make: *** [Makefile:43: validate-components] Error 1

Above you can see a failure to build the Contrib distributions.

This happens even in completely unrelated PRs, like #1209. See the full build log at https://github.com/open-telemetry/opentelemetry-collector-releases/actions/runs/18555999587/job/52893535451?pr=1209 for more details.

--- a/scripts/validate-components.sh
+++ b/scripts/validate-components.sh
@@ -32,9 +32,24 @@ CURL_FETCH_ARGS=(
 
 # Fetch and parse valid components from builder-config.yaml
 echo "Fetching builder-config.yaml..."
+builder_config_tmp="$(mktemp)"
+trap 'rm -f "$builder_config_tmp"' EXIT
+if ! http_status="$(
+  curl "${CURL_FETCH_ARGS[@]}" \
+    --write-out '%{http_code}' \
+    --output "$builder_config_tmp" \
+    "$BUILDER_CONFIG_URL"
+)"; then
+  echo "Failed to fetch builder-config.yaml via curl." >&2
+  exit 1
+fi
+if [[ "$http_status" != "200" ]]; then
+  echo "Failed to fetch builder-config.yaml: HTTP ${http_status}" >&2
+  exit 1
+fi
+
 valid_components="$(
-  curl "${CURL_FETCH_ARGS[@]}" "$BUILDER_CONFIG_URL" \
-    | yq -r '
+  yq -r '
       (
         .extensions[]?.gomod,
         .receivers[]?.gomod,
@@ -43,7 +58,7 @@ valid_components="$(
         .exporters[]?.gomod,
         .providers[]?.gomod
       )
-    ' \
+    ' "$builder_config_tmp" \
     | awk '{print $1}' \
     | sort -u
 )"

